engine/consensus/sealing/core.go

```diff
@@ -241,16 +247,25 @@ func (c *Core) RepopulateAssignmentCollectorTree(payloads storage.Payloads) erro
 // * exception in case of any other error, usually this is not expected
 // * nil - successfully processed incorporated result
 func (c *Core) processIncorporatedResult(incRes *flow.IncorporatedResult) error {
-	err := c.checkBlockOutdated(incRes.Result.BlockID)
-	if err != nil {
-		return fmt.Errorf("won't process outdated or unverifiable execution incRes %s: %w", incRes.Result.BlockID, err)
-	}
+
 	incorporatedBlock, err := c.headers.ByBlockID(incRes.IncorporatedBlockID)
 	if err != nil {
 		return fmt.Errorf("could not get block height for incorporated block %s: %w",
 			incRes.IncorporatedBlockID, err)
 	}
 	incorporatedAtHeight := incorporatedBlock.Height
+	belowRootHeight := incorporatedAtHeight <= c.rootHeader.Height
+
+	err = c.checkBlockOutdated(incRes.Result.BlockID)
+	// we can safely ignore results incorporated below the root height which
+	// are unverifiable (reference unknown blocks) - these are blocks which
+	// are incorporated prior to this node's initial snapshot
+	if belowRootHeight && engine.IsUnverifiableInputError(err) {
```

Comment:
It seems to me it's safe to check only if it's belowRootHeight and just drop it, we don't even need to check if block is outdated since it's already below root snapshot height == outdated

Reply:
I wasn't sure about this point. 

When we initialize a new node with a sealing segment, we have an initial state like:
```
A<-B<-C<-D
         ^- root height / finalized height
^-sealed height
```

The node only knows that `A` is sealed, so information about sealing `B`, `C`, and `D` is still (at least theoretically) relevant to us. Practically, with the joining process we've defined, the node would not actively participate in sealing these blocks because the node would have 0 weight at this time (they would observe the sealing by incorporating blocks proposed by others, but would not propose or vote on these blocks). 

I feel it is safer to be strict about this here, and make sure we initialize the sealing engine with ERs for unsealed blocks in this case as well.

Reply:
Does sealing engine need to process those result at all if the block of the result is unknown?

To me it's an overkill to adjust the logic in `processIncorporatedResult`. `processIncorporatedResult` should just assume the block of the input result must be found. Adding the logic here makes this function more complex.

For such a sealing segment, we can create a list of unknown blocks referred by all results from the blocks in the sealing segment. And not to process them [when initializing the sealing core here](https://github.com/onflow/flow-go/blob/139cf03b1c9971af34f39e22a5e1c45a90724a9e/engine/consensus/sealing/core.go#L178-L179)
```
A<-B(Result_X)<-C<-D
```

Once the sealing core has been initialized, all future incorporated results must refer to a known block. 

So I think we should update the `RepopulateAssignmentCollectorTree` function, rather than the `processIncorporatedResult`

Reply:
> I feel it is safer to be strict about this here, and make sure we initialize the sealing engine with ERs for unsealed blocks in this case as well. But let me know what you think.

That is actually right, I've thought about root height a little bit differently. Yeah, better to be strict.

@zhangchiqing 's suggestion is also worth thinking about. Indeed this happens only when initialization, so maybe we deal with that explicitly in `RepopulateAssignmentCollectorTree` ? Later we can't even receive such result since it must be incorporated into some block.

Reply:
👍  Going to go ahead with Leo's suggestion